[Merge] HC-92-Avatar-component to dev

siuhyun opened the pull request:
## 📝 개요(이미지 첨부 선택)
*<!-- 이슈에 대해 어떻게, 무엇을, 왜 수정했는지 작성해주세요. -->*
### Atom Component Avatar 생성 
> 사이즈별 분류 단위
- XS; 40px
    - chatRoom 
- S; 44px:
    - new notice dropdown
- M; 48px:
    - chatList Avatar
- L; 60px:
    - house detail page 댓글
- XL; 72px
    - RoomMate Application Status Modal
    - user dropdown
    - house detail page
- XXL; 96px:
    - mypage 내활동
- XXXL; 128px:
    - mypage 내활동 → 계정설정 Avatar

### Avatar 사용 예시
```tsx
// 일반 상태
<Avatar.XS src="https://i.pravatar.cc/300?img=1" />
// Active 상태
<Avatar.XS isActive src="https://i.pravatar.cc/300?img=1" />
```
> 상태에 따른 스타일
```tsx
boxShadow: {
        avatar: 'rgba(0, 0, 0, 0.25) 0px 0px 4px 0px',
        'avatar-active': 'rgba(0, 0, 0, 0.3) 0px 0px 4px 3px',
      },
```

<img width="889" alt="스크린샷 2024-07-07 오전 12 07 19" src="https://github.com/house-mate-connect/house-connect/assets/149446617/27a8b4fb-7eb6-4def-943a-79b9f936d2da">


## ✅ PR 유형
어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [x] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [x] 주석 추가 및 수정
- [ ] 문서 수정
- [x] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## 📌 PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.  
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).
- [ ] 모든 팀원이 PR에 대한 리뷰를 진행했습니다.
- [ ] 이 코드 변경 사항은 기존 기능에 영향을 주지 않나요?
- [ ] 팀원이 이해할 수 있도록 충분한 주석 및 코드 일관성을 준수했나요? (변수명, 함수명, 클래스명, 가독성 등)

## 💁‍♂️ PR 추가 정보
*<!-- PR과 관련된 기타 정보(E.G 개선사항, 리뷰어가 봐주었으면 하는 점)가 있다면 여기에 적어주세요. -->*
Avatar size 분류명 추후 수정 필요해보입니다.
active 상태 스타일도 추후 수정 필요해보입니다. 
아이디어 있으면 남겨 주세요. 🫡

siuhyun committed:
🚀 [Feat]: Avatar Component 생성

- 사이즈 별 분류: XS, S, M, L, XL, XXL, XXXL
- isActive 상태에 따라 추가 스타일 shadow-avatar-active 적용
- 'avatar-active': 'rgba(0, 0, 0, 0.3) 0px 0px 4px 3px'
- ComponentTest 페이지에 Avatar Sample  생성

rlwjd31 commented on src/components/atoms/Avatar.tsx:
```diff
@@ -0,0 +1,72 @@
+    - mypage 내활동
+- 3xl; 128px:
+    - mypage 내활동 → 계정설정 Avatar
+*/
```
docs 너무 맘에 듬!

rlwjd31 commented on src/components/atoms/Avatar.tsx:
```diff
@@ -0,0 +1,72 @@
+    ...others
+  }: AvatarProps) =>
+    React.createElement('img', {
+      className: `${defaultClassName} ${cn('shadow-avatar shrink-0 cursor-pointer rounded-full', className, isActive && 'shadow-avatar-active')}`,
```
- defaultClassName은 style이 무조건 적용되어야 하므로 cn함수의 맨 마지막으로 이동해야 할 듯
- 나머지 style은 cn함수에 ,를 이용하여 전개부탁

rlwjd31 commented on src/components/atoms/Avatar.tsx:
```diff
@@ -0,0 +1,72 @@
+const Avatar = {} as AvatarComponentProps;
+AvatarSize.forEach(({ size, defaultClassName }) => {
+  Avatar[size] = ({
+    children,
```
img element는 self closing element로 children이 필요없음

rlwjd31 commented on src/components/pages/ComponentTest.tsx:
```diff
@@ -587,6 +588,29 @@ export default function ComponentTest() {
+      <br />
+      <h3>Avatar Active 상태</h3>
+      <Container.FlexRow className="w-full items-center gap-3">
+        <Avatar.XS isActive src="https://i.pravatar.cc/300?img=1" />
```
isActive값은 마우스가 active된 상태임을 의미하는 거 같은데
isActive props를 update하기 위한 로직이 없음

- tailwind: "active:shadow-avatar-active"가 되어야 맥락이 맞을 듯 함.

kwondns replied:
active가 css 레벨의 active가 아니라 드랍박스 활성화 되어있을때에 대한 active 아닌지?

rlwjd31 commented on tailwind.config.ts:
```diff
@@ -57,6 +57,7 @@ export default {
       },
       boxShadow: {
         avatar: 'rgba(0, 0, 0, 0.25) 0px 0px 4px 0px',
+        'avatar-active': 'rgba(0, 0, 0, 0.3) 0px 0px 4px 3px',
```
기존에 있는 avatar는 sample용이라 지우고 avatar라는 이름으로 재정의 하면 됨

kwondns replied:
avatar는 기본 상태고 active 상태일 때 별도로 정의하는거같은데

New-Ha reviewed (approved):
👍🏻 수정사항만 하시면 아바타 부분은 다 커버할 것 같아요!

siuhyun committed:
♻️ [Refactor]:  PR 리뷰 수정사항

- Style 적용 우선순위 변경
- defaultClassName cn함수 마지막으로 이동
- children Props 제거

Resolves: #51

kwondns reviewed (approved):
ad6c15a8555b3d589ae08c683431c4280db799a5 
커밋으로 수정사항 확인 완료
